css-scroll-snap-2/Overview.bs

```diff
@@ -275,119 +275,209 @@ Snap Events {#snap-events}
+			<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+			1. Append |snapcontainer| to <var>doc</var>'s
+				<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+	1. If <var>inlineTarget</var> is not the same element as
+			<var>inlineSnapchangingTarget</var>:
+		1. Set the <a>snapchangedTargetInline</a> associated with |snapcontainer| to
+			<var>inlineSnapchangingTarget</var>.
+		1. If |snapcontainer| is not already in <var>doc</var>'s
+			<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+			1. Append |snapcontainer| to <var>doc</var>'s
+				<a spec=cssom-view lt="pending snapchanged event targets">pending snapchanged event targets</a>.
+
+<h4 id="snapchanging"> snapchanging </h4>
+	{{snapchanging}} is dispatched when the element to which a
+	snap container would snap (in either axis) during a scrolling operation changes.
+	A scrolling operation may animate towards a particular position (e.g.
+	scrollbar arrow clicks, arrow key presses, "behavior: smooth" programmatic
+	scrolls) or may directly track a user's input (e.g. touch scrolling, scrollbar
+	dragging).
+	* In the former case, the user agent should evaluate whether snapchanging is
+		occurring based on the
+		<a spec="css-scroll-snap-1" lt="scroll snap position">snap position</a>
+		that will eventually be snapped to after the scroll animation's target offset
+		is reached.
+	* In the latter case, the user agent should evaluate whether snapchanging
+		should be triggered based on the
+		<a spec="css-scroll-snap-1" lt="scroll snap position">snap position</a> that
+		would be snapped to if the scroll is ended at the current scroll position
+		(determined by the user's input).
+
+	{{snapchanging}} aims to let the web page know, as early as possible,
+	that the scrolling operation will result in a change in the element the snap
+	container is snapped to.
+
+	{{snapchanging}} events should fire before {{scroll}} events.
+
+	Note: Since snapchanging gives the web page hints about future snapping,
+	the snapping hinted at by a snapchanging event may not materialize since it
+	will be	possible for subsequent scrolling input to further alter the snap
+	container's scroll position and result in a different eventual snap position.
+
+	Each
+	<a spec=css-scroll-snap lt="scroll snap container">snap container</a> is
+	associated with at most one <dfn>snapchangingTargetBlock</dfn> and at most one
+	<dfn>snapchangingTargetInline</dfn> in the block and inline axes respectively.
+
+	When asked to <dfn export for=Document>update snapchanging targets</dfn>
+	for a <a spec=css-scroll-snap lt="scroll snap container">snap container</a>,
+	|snapcontainer|, run these steps:
+
+	1. Let <var>doc</var> be |snapcontainer|'s associated {{Document}}.
+	1. Let <var>scrollPosition</var> be the scroll offset resulting from applying
+		all pending scroll updates to |snapcontainer|.
```

Comment:
I don't think "pending scroll updates" is quite correct as they're not pending yet. I think you should define a term such as "eventual snap target" above where you describe the cases of scrolling operations, and then refer to that term here.